pandas/tests/test_resample.py

```diff
@@ -2252,12 +2278,17 @@ def test_selection(self):
                           index=pd.MultiIndex.from_arrays([
                               np.arange(len(index), dtype=np.int64),
                               index], names=['v', 'd']))
-
-        with pytest.raises(NotImplementedError):
-            df.resample('2D', on='date')
-
-        with pytest.raises(NotImplementedError):
-            df.resample('2D', level='d')
+        for freq in ['H', '12H', '2D', 'W']:
```

Comment:
these tests are more clear with parametrize 
but we still inherit from TestCase so can't change this

but you can start a new test class that does this 

again if it's more clear